stdlib/src/builtin/builtin_list.mojo

```diff
@@ -80,6 +85,74 @@ struct ListLiteral[*Ts: CollectionElement](Sized, CollectionElement):
         """
         return len(self.storage)
 
+    fn __str__(self) -> String:
+        """Returns a string representation of a ListLiteral.
+
+        Returns:
+            The string representation of the ListLiteral.
+
+        Here is an example below:
+        ```mojo
+        var my_list = [1, 2, "Mojo"]
+        print(str(my_list))
+        ```.
+        """
+        var output = String()
+        var writer = output._unsafe_to_formatter()
+        self.format_to(writer)
+        return output^
+
+    fn __repr__(self) -> String:
+        """Returns a string representation of a ListLiteral.
+
+        Returns:
+            The string representation of the ListLiteral.
+
+        Here is an example below:
+        ```mojo
+        var my_list = [1, 2, "Mojo"]
+        print(repr(my_list))
+        ```.
+        """
+        return self.__str__()
+
+    @no_inline
+    fn format_to(self, inout writer: Formatter):
+        """Format the list literal.
+
+        Args:
+            writer: The formatter to use.
+        """
+        writer.write("[")
+
+        @parameter
+        for i in range(len(VariadicList(Ts))):
+            alias T = Ts[i]
+            var s: String
+            if i > 0:
+                writer.write(", ")
+
+            @parameter
+            if _type_is_eq[T, PythonObject]():
```

Comment:
@jackos would this sort of thing able to be improved with your in-flight PR for writer?  I don't love explicitly enumerating a bunch of types that we know are convertible to writable since we just store a `Tuple[*Ts]` and don't know if the `Ts` are writable themselves.